src/compiler/checker.ts

```diff
@@ -573,7 +573,16 @@ namespace ts {
                         declarationNameToString(propertyName), typeof nameArg === "string" ? nameArg : declarationNameToString(nameArg));
                     return undefined;
                 }
-                if (result.flags & SymbolFlags.BlockScopedVariable) {
+
+                // Only check for block-scoped variable if we are looking for the name with variable meaning
+                // For example,
+                //      declare module foo {
+                //          interface bar {}
+                //      }
+                // let foo/*1*/: foo/*2*/.bar;
+                // The foo at /*1*/ and /*2*/ will share same symbol with two meaning block-scope variable and namespace module.
+                // However, only when we try to resolve name in /*1*/ which is used in variable position, we want to check for block- scoped

```

Comment:
I agree about the assertion but I don't understand how the assertion convey similar meaning with the comment. ( I don't mind remove it)